modin/experimental/core/execution/native/implementations/omnisci_on_native/exchange/dataframe_protocol/column.py

```diff
@@ -0,0 +1,488 @@
+        elements_in_buffer = self.size + 1 if is_offset_buffer else self.size
+        result = ceil((bit_width * elements_in_buffer) / 8)
+        # For a bitmask, if the chunk started in the middle of the byte then we need to
+        # read one extra byte from the buffer to retrieve the chunk's tail in the last byte. Example:
+        # Bitmask of 3 bytes, the chunk offset is 3 elements and its size is 16
+        # |* * * * * * * *|* * * * * * * *|* * * * * * * *|
+        #      ^- the chunk starts here      ^- the chunk ends here
+        # Although ``ceil(bit_width * elements_in_buffer / 8)`` gives us '2 bytes',
+        # the chunk is located in 3 bytes, that's why we assume the chunk's buffer size
+        # to be 'result += 1' in this case:
+        if bit_width == 1 and self.offset % 8 + self.size > result * 8:
+            result += 1
+        return result
+
+    def _get_data_buffer(
+        self, arr: pa.Array
+    ) -> Tuple[OmnisciProtocolBuffer, Tuple[DTypeKind, int, str, str]]:
+        """
+        Get column's data buffer.
+
+        Parameters
+        ----------
+        arr : pa.Array
+            PyArrow array holding column's data.
+
+        Returns
+        -------
+        tuple
+            Tuple of OmnisciProtocolBuffer and protocol dtype representation of the buffer's underlying data.
+        """
+        if self.dtype[0] == DTypeKind.CATEGORICAL:
+            # For dictionary data the buffer has to return categories codes
+            arr = arr.indices
+
+        arrow_type = self._dtype_from_pyarrow(arr.type)
+        buff_size = (
+            self._get_buffer_size(bit_width=arrow_type[1])
+            if self.dtype[0] != DTypeKind.STRING
+            # We don't chunk string buffers as it would require modifying offset values,
+            # so just return the whole data buffer for every chunk.
+            else None
+        )
+
+        return (
+            OmnisciProtocolBuffer(arr.buffers()[-1], buff_size),
+            arrow_type,
+        )
+
+    def _get_validity_buffer(
+        self, arr: pa.Array
+    ) -> Optional[Tuple[OmnisciProtocolBuffer, Tuple[DTypeKind, int, str, str]]]:
+        """
+        Get column's validity buffer.
+
+        Parameters
+        ----------
+        arr : pa.Array
+            PyArrow array holding column's data.
+
+        Returns
+        -------
+        tuple or None
+            Tuple of OmnisciProtocolBuffer and protocol dtype representation of the buffer's underlying data.
+            None if column is non-nullable (``self.describe_null == ColumnNullType.NON_NULLABLE``).
+        """
+        validity_buffer = arr.buffers()[0]
+        if validity_buffer is None:
+            return validity_buffer
+
+        # If exist, validity buffer is always a bit-mask.
+        data_size = self._get_buffer_size(bit_width=1)
+        return (
+            OmnisciProtocolBuffer(validity_buffer, data_size),
+            (DTypeKind.BOOL, 1, "b", "="),
```

Comment:
The bitmask for bool is not equal to 1.

Reply:
You mean that `bitwidth` for a bit-mask should be `8` and the buffer itself has to be interpreted as a `uint8`? Why would we do that? The consumer already has to handle boolean columns as a pure bit-mask, why do we need different logic for validity buffer?

Reply:
Given [this](https://github.com/modin-project/modin/pull/4269#discussion_r826130829) the bitwidth is apparently equal to 1. Let's leave as is